packages/ui/src/elements/Upload/index.tsx

```diff
@@ -173,19 +185,49 @@ export const Upload: React.FC<UploadProps> = (props) => {
   )
 
   const handleUrlSubmit = async () => {
+    // Disable URL submission if pasteURL is false
+    if (uploadConfig?.pasteURL === false) {
+      return
+    }
+
     if (fileUrl) {
       setUploadStatus('uploading')
-      try {
-        const response = await fetch(fileUrl)
-        const data = await response.blob()
-
-        // Extract the file name from the URL
-        const fileName = fileUrl.split('/').pop()
 
-        // Create a new File object from the Blob data
-        const file = new File([data], fileName, { type: data.type })
-        handleFileChange(file)
-        setUploadStatus('idle')
+      try {
+        // Check if pasteURL.allowList is defined
+        if (useServerSideFetch) {
+          // Perform server-side fetch if the URL is allowed
+          if (isUrlAllowed(fileUrl, uploadConfig.pasteURL.allowList)) {
+            const pasteURL = `/${collectionSlug}/paste-url${id ? `/${id}?` : '?'}src=${encodeURIComponent(fileUrl)}`
+            const response = await fetch(`${serverURL}${api}${pasteURL}`)
+
+            if (!response.ok) {
+              const errorData = await response.json()
+              throw new Error(errorData.error || 'Failed to fetch the file from the server')
+            }
+
+            const blob = await response.blob()
+            const fileName = fileUrl.split('/').pop()
+            const file = new File([blob], fileName, { type: blob.type })
+
+            handleFileChange(file)
+            setUploadStatus('idle')
+          } else {
+            throw new Error('The provided URL is not allowed.')
+          }
+        } else {
+          // Perform client-side fetch if no allowList is specified
+          const response = await fetch(fileUrl)
```

Comment:
I was thinking that this feature would first attempt to make the fetch client side, without checking the URL at all.
If the client side fetch fails and the pasteURL.allowList is defined then we would make the server side call to getFileFromURL.

What do you think?